bot/exts/filtering/_settings_types/actions/infraction_and_notification.py

```diff
@@ -0,0 +1,203 @@
+    WARNING = auto()
+    WATCH = auto()
+    NOTE = auto()
+    NONE = auto()
+
+    def __str__(self) -> str:
+        return self.name
+
+    async def invoke(
+        self,
+        user: Member | User,
+        channel: discord.abc.GuildChannel | discord.DMChannel,
+        alerts_channel: discord.TextChannel,
+        duration: float,
+        reason: str
+    ) -> None:
+        """Invokes the command matching the infraction name."""
+        command_name = self.name.lower()
+        command = bot_module.instance.get_command(command_name)
+        if not command:
+            await alerts_channel.send(f":warning: Could not apply {command_name} to {user.mention}: command not found.")
+            log.warning(f":warning: Could not apply {command_name} to {user.mention}: command not found.")
+            return
+
+        if isinstance(user, discord.User):  # For example because a message was sent in a DM.
+            member = await get_or_fetch_member(channel.guild, user.id)
+            if member:
+                user = member
+        ctx = FakeContext(channel, command)
+        if self.name in ("KICK", "WARNING", "WATCH", "NOTE"):
+            await command(ctx, user, reason=reason or None)
+        else:
+            duration = arrow.utcnow() + timedelta(seconds=duration) if duration else None
+            await command(ctx, user, duration, reason=reason or None)
+
+
+class InfractionAndNotification(ActionEntry):
+    """
+    A setting entry which specifies what infraction to issue and the notification to DM the user.
+
+    Since a DM cannot be sent when a user is banned or kicked, these two functions need to be grouped together.
+    """
+
+    name: ClassVar[str] = "infraction_and_notification"
+    description: ClassVar[dict[str, str]] = {
+        "infraction_type": (
+            "The type of infraction to issue when the filter triggers, or 'NONE'. "
+            "If two infractions are triggered for the same message, "
+            "the harsher one will be applied (by type or duration).\n\n"
+            "Valid infraction types in order of harshness: "
+        ) + ", ".join(infraction.name for infraction in Infraction),
+        "infraction_duration": "How long the infraction should last for in seconds. 0 for permanent.",
+        "infraction_reason": "The reason delivered with the infraction.",
+        "infraction_channel": (
+            "The channel ID in which to invoke the infraction (and send the confirmation message). "
+            "If 0, the infraction will be sent in the context channel. If the ID otherwise fails to resolve, "
+            "it will default to the mod-alerts channel."
+        ),
+        "dm_content": "The contents of a message to be DMed to the offending user.",
+        "dm_embed": "The contents of the embed to be DMed to the offending user."
+    }
+
+    dm_content: str
+    dm_embed: str
+    infraction_type: Infraction
+    infraction_reason: str
+    infraction_duration: float
```

Comment:
Passing the infraction duration is quite confusing, it would be nice if it is taken/displayed using the normal infraction duration format e.g. `1d`, currently you just get a message saying "Could not process the input value for infraction_duration.".

Another option would be if the "How long the infraction should last for in seconds. 0 for permanent." info could be put in the label of the modal.

Reply:
I wish I could put it in the modal, but the lengths of everything are pretty limiting. The title and any labels are up to 45 chars, placeholder strings are up to 100. I can make the description the default value but then people will need to delete it every time they want to input a value, which seems bad. I'll see if I can come up with something decent.

Reply:
This should do it
[Support custom value representation if filtering UI](https://github.com/python-discord/bot/pull/2390/commits/5850002237293be3ba0c2f9fd3ce707303a47679)

Reply:
Much better, thanks!